activemodel/lib/active_model/validator.rb

```diff
@@ -100,7 +100,7 @@ class Validator
     #   PresenceValidator.kind   # => :presence
     #   UniquenessValidator.kind # => :uniqueness
     def self.kind
-      @kind ||= name.split('::').last.underscore.sub(/_validator$/, '').to_sym unless anonymous?
+      @kind ||= name.split('::').last.underscore.chomp('_validator').to_sym unless anonymous?
```

Comment:
Yes, I agree, it's a micro optimization. However it's kind of example. If it's OK, I could refactor code in other places, where it can be more useful